### PR TITLE
Improve error handling for export encoding

### LINKED ISSUE #888: TypeError: Cannot read property '1' of null
Sentry Issue: [KAP-5KT](https://sentry.io/organizations/wulkano-l0/issues/1532786702/?referrer=github_integration)

```
TypeError: Cannot read property '1' of null
  File "app:///main/utils/encoding.js", line 17, in getEncoding
    return /.*: Video: (.*?) \(.*/.exec(error.stderr)[1];
```

### PR DESCRIPTION
**Pull request by andreidobrinski**
Fixes #888 

Hope this helps!

### CODE REVIEW

**Review by sindresorhus — APPROVED** [review]

**Comment by virtuoushub** [comment]
@andreidobrinski Just need to remove the whitespace from the updated file.
```patch
From 7dc60d46521f7be77c022dc15c23ebde18ab33b4 Mon Sep 17 00:00:00 2001
From: Peter Colapietro <petercolapietro@gmail.com>
Date: Fri, 2 Oct 2020 08:33:28 -0400
Subject: [PATCH] style(whitespace): fix whitespace errors

breaking CI
See: https://app.circleci.com/pipelines/github/wulkano/Kap/168/workflows/047525cc-f95b-41aa-bf6e-be40c377aa7c/jobs/1721/parallel-runs/0/steps/0-103
---
 main/utils/encoding.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/main/utils/encoding.js b/main/utils/encoding.js
index 3c48803..0307d57 100644
--- a/main/utils/encoding.js
+++ b/main/utils/encoding.js
@@ -15,11 +15,11 @@ const getEncoding = async filePath => {
     await execa(ffmpegPath, ['-i', filePath]);
   } catch (error) {
     const errorText = /.*: Video: (.*?) \(.*/.exec(error.stderr);
-    
+
     if (!errorText) {
       throw error;
     }
-    
+
     return errorText[1];
   }
 };
-- 
2.27.0
```

**Comment by andreidobrinski** [comment]
Thanks @virtuoushub and @sindresorhus for the feedback.

I updated the PR to remove whitespace.

Thanks also for your work, I love the app!

File: main/utils/encoding.js [hunk 1]
```diff
@@ -14,7 +14,11 @@ const getEncoding = async filePath => {
   try {
     await execa(ffmpegPath, ['-i', filePath]);
   } catch (error) {
-    return /.*: Video: (.*?) \(.*/.exec(error.stderr)[1];
+    const errorText = /.*: Video: (.*?) \(.*/.exec(error.stderr);
+    if (!errorText) {
+      throw error;
+    }
+    return errorText[1];
```
Comment (virtuoushub) [inline main/utils/encoding.js:21]: @andreidobrinski, almost there. Looking at the [CI's error output](https://app.circleci.com/pipelines/github/wulkano/Kap/169/workflows/b44a6fd4-d110-464b-88e6-4f0952cddc63/jobs/1722), I believe the linter wants:
```javascript
    const errorText = /.*: Video: (.*?) \(.*/.exec(error.stderr);
    if (!errorText) {
      throw error;
    }

    return errorText[1];
```

We can use `yarn test` locally to verify.

**Review by virtuoushub — CHANGES_REQUESTED** [review]

File: main/utils/encoding.js (hunk 1 above)
Comment (andreidobrinski) [reply main/utils/encoding.js:22]: Thanks for the tip! I committed the update :)

**Comment by sindresorhus** [comment]
Still failing linting. Please always run `npm test` locally after doing changes.

**Comment by andreidobrinski** [comment]
Should be ok now. Thanks!

**Review by Ayhanbey63500 — APPROVED** [review]
``main/utils/encoding.jsmain/utils/encoding.js

**Comment by sindresorhus** [comment]
https://github.com/wulkano/Kap/issues/888#issuecomment-1208771208

**Pull request closed without merging**